src/chainlit/socket.py

```diff
@@ -158,13 +160,9 @@ async def process_message(session: Session, author: str, input_str: str):
 
         if session.db_client:
             # If cloud is enabled, persist the message
-            await session.db_client.create_message(
-                {
-                    "author": author,
-                    "content": input_str,
-                    "authorIsUser": True,
-                }
-            )
+            await session.db_client.create_message(**message)
+
+        session.last_message = Message.from_dict(message)
```

Comment:
I think we should also set last_message on the .send method of the Message class

Reply:
The point of that variable is to keep track of the current processing head, so I don't think we want to overwrite it with the messages we send. That being said, I guess its current name is quite poor and we could rename it with something like `last_user_message`, `last_input`, `current_user_message` or anything else. Ideas welcome! :)

Reply:
I don't think this is necessarily the last user message.

Let's consider this small chainlit app:

```py
import chainlit as cl


@cl.on_chat_start
async def start():
    await cl.Message(
        content="Intermediary steps should be attached on this message",
    ).send()
    
    # Run my langchain agent ...
```

Here no user message is every sent, but we still need to attach the intermediary step messages created by the callback to the "last message sent", in this case the message sent in `on_chat_start`.

Reply:
as discussed, let's always use the last root message or the root message we provide to the callback handler